### ISSUE TITLE
Deleting an ACL of a deleted consumer results in a HTTP 500 - an unexpected error occurred

### ISSUE THREAD

**Issue by mvanholsteijn** [Kong/kong#10669]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

2.8.3

### Current Behavior

When I :
- create a consumer 
- add an ACL to the consumer
- delete the consumer
- delete the ACL

The ACL delete causes a server error (500).

The expected behaviour is to receive a 404, as I get when I attempt to delete an already deleted consumer.

### Expected Behavior

When I :
- create a consumer 
- add an ACL to the consumer
- delete the consumer
- delete the ACL

The delete ACL returns a 404.

### Steps To Reproduce

Run the following scripts against a test kong instance:

```bash
#!/bin/bash
echo "## delete the consumer"
curl -X DELETE http://localhost:8001/consumers/deleteme
echo ""

echo "## create the consumer"
CONSUMER_ID=$(curl -sS -X POST http://localhost:8001/consumers --data username=deleteme | jq -r .id)
echo ""

echo "## add the ACL to the consumer"
ACL_ID=$(curl -sS -X POST http://localhost:8001/consumers/deleteme/acls --data group=admin | jq -r .id)
echo ""

echo "## delete the consumer"
curl  -X DELETE http://localhost:8001/consumers/$CONSUMER_ID
echo ""

echo "## delete the acl"
curl  -X DELETE http://localhost:8001/consumers/$CONSUMER_ID/acls/$ACL_ID
echo ""
```
In de case of version 2.8 the last delete will read:

```
## delete the acl
{"message":"An unexpected error occurred"}
```
With version 3.2.2 the message is:
```
## delete the acl
{"message":"Not found"}
```

**Comment by attenuation** [comment]
I'm sorry, but according to our policy, version 2.8 of the community edition is not supported. If you want to resolve this issue, we recommend upgrading to the latest version of Kong.

**Comment by mvanholsteijn** [comment]
Will do. Just wanted to make sure you knew about this problem,

**Comment by mvanholsteijn** [comment]
Solved in 3.2.2
